js-plot/README.md

```diff
@@ -2,22 +2,214 @@
 
 Running C++ code in a web browser is all nice, but we really want to grab someones attention by visualizing something. In this blog we are going to make a plot from the results coming from our C++ code.
 
-To make a plot we need some data. In a [previous](../js-webapp) post we found the root of an equation using the Newton-Raphson algorithm implemented in C++ and compiled to a WebAsssembly module.
+To make a plot we need some data. In a [previous](TODO fix url ../js-webapp) post we found the root of an equation using the Newton-Raphson algorithm implemented in C++ and compiled to a WebAsssembly module.
 A single root value makes for a depressing plot. The Newton-Raphson algorithm uses iterations to find the root so we will capture the data of each iteration and plot those.
 
 Let's make changes to the C++ code to retrieve the data from the iterations.
 
-TODO add code snippets
+## Iterations
 
-TODO explain how wasm vector is converted to array
+We will add `iterations` public property to the NewtonRaphson which is a vector of iteration structs. So the `newtonraphson.hpp` becomes
 
-TODO show iterations from intiaizas json doc
+```cpp
+#ifndef H_NEWTONRAPHSON_H
+#define H_NEWTONRAPHSON_H
+
+#include <vector>
+
+struct Iteration {
+  int index;
+  double x;
+  double y;
+  double slope;
+  double delta_x;
+};
+
+class NewtonRaphson {
+  public:
+    NewtonRaphson(double tolerancein);
+    double solve(double xin);
+    std::vector<Iteration> iterations;
+  private:
+    double tolerance;
+};
+#endif
+```
+File: _newtonraphson.hpp_
+
+The `newtonraphson.cpp` is rewritten from a while loop to a while/do loop like with a push to the iterations vector each cycle.
```

Comment:
It is strange to see this. Shall we also change the initial loop to have do-while?

Reply:
fine by me, but lets do it in another PR